OsmAnd/src/net/osmand/plus/views/AsynchronousResampler.java

```diff
@@ -0,0 +1,282 @@
+package net.osmand.plus.views;
+
+import android.graphics.Color;
+import android.os.AsyncTask;
+
+import net.osmand.plus.GPXUtilities;
+import net.osmand.plus.views.OsmandMapTileView;
+import net.osmand.util.MapUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class AsynchronousResampler extends AsyncTask<String,Integer,String> {
+
+    private OsmandMapTileView view;
+    private Renderable.RenderableSegment rs;
+    private Renderable.RenderType renderType;
+    private double param1, param2;
+
+    private List<GPXUtilities.WptPt> culled = null;
+
+
+    public AsynchronousResampler(Renderable.RenderType rt, OsmandMapTileView view,
+                                 Renderable.RenderableSegment rs, double param1, double param2) {
+        this.view = view;
+        this.rs = rs;
+        this.renderType = rt;
+
+        this.param1 = param1;
+        this.param2 = param2;
+    }
+
+    public int getColor2(double percent) {
+
```

Comment:
OK, I will do that. I have already rewritten it to look nice.
Essentially it will, given an input between 0 and 1 (double), return a colour of the rainbow from purple to red.
Great for colour-coding a range of values such as altitude or speed.

> It is better to have that piece of code somewhere in Algorithms class with proper explanation what is doing it looks very generic.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/osmandapp/Osmand/pull/2379/files/109ca692f100ddb59f0e538b06c32beae5910bc0#r57644979>